digicampipe/scripts/pulse_shape.py

```diff
@@ -1,78 +1,121 @@
 #!/usr/bin/env python
-'''
-Reconstruct the pulse template
-
+"""
+Create an histogram of the pulse template for each pixel.
 Usage:
   digicam-template [options] <input_files>...
 
 Options:
-  -h --help               Show this screen.
-  --output=PATH           outfile path, if not given, we just append ".h5" to
-                          the input file path
-'''
-import h5py
+  -h --help                 Show this screen.
+  --output_hist=PATH        Output histogram file, if not given, we just append
+                            ".npz" to the path of the 1st input file.
+  --time_range_ns=LIST      Minimum and maximum time in ns w.r.t. half maximum
+                            of the pulse during rise time [default: -10,40].
+  --amplitude_range=LIST    Minimum and maximum amplitude of the template
+                            normalised in integrated charge [default: -.1,0.4].
+  --integration_range=LIST  Minimum and maximum indexes of samples used in the
+                            integration for normalization of the pulse charge
+                            [default: 8,20].
+  --charge_range=LIST       Minimum and maximum integrated charge in LSB used
+                            to build the histogram [default: 1000,10000].
+  --n_bin=INT               Number of bins for the 2d histograms
+                            [default: 100].
+
+"""
 import numpy as np
 from docopt import docopt
-from tqdm import tqdm
+import os
 
-from digicampipe.calib.time import estimate_arrival_time
+from digicampipe.calib.time import estimate_time_from_leading_edge
 from digicampipe.io.event_stream import calibration_event_stream
 from digicampipe.utils.hist2d import Histogram2dChunked
-
-
-def main(outfile_path, input_files=[]):
+from digicampipe.calib.baseline import fill_digicam_baseline, \
+    subtract_baseline, correct_wrong_baseline
+from digicampipe.utils.docopt import convert_list_float, convert_list_int, \
+    convert_int
+
+
+def main(
+        output_hist,
+        input_files,
+        time_range_ns=(-10., 40.),
+        amplitude_range=(-0.1, 0.4),
+        integration_range=(8, 20),
+        # charge < 50 pe (noisy) or > 500 pe (saturation) => bad_charge
+        # 1 pe <=> 20 LSB integral
+        charge_range=(1000., 10000.),
+        n_bin=100,
+):
+    if os.path.exists(output_hist):
+        os.remove(output_hist)
+    charge_min = np.min(charge_range)
+    charge_max = np.max(charge_range)
+    integration_min = np.min(integration_range)
+    integration_max = np.max(integration_range)
     events = calibration_event_stream(input_files)
-    Rough_factor_between_single_pe_amplitude_and_integral = 21 / 5.8
+    events = fill_digicam_baseline(events)
+    if "SST1M_01_201805" in input_files[0]:  # fix data in May
+        print("WARNING: correction of the baselines applied.")
```

Comment:
maybe we could put a "real" warning in the reader not necessary in this PR

Reply:
There are data in May which are not affected, so that simple check would produce bad result if put in the reader.
The version of digicam/camera server was not yet present in that time in the header of the zfits, so we cannot use that.
If I knew how to do it in a proper way, I would indeed have moved it to the reader.